packages/core-data/src/hooks/use-entity-record.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * WordPress dependencies
+ */
+import { useQuerySelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as coreStore } from '../';
+import { Status } from './constants';
+
+interface EntityRecordResolution {
+	/** The requested entity record */
+	record: Object;
+
+	/** The requested entity record with any edits applied*/
+	editedRecord: Object;
+
+	/** Is the record still being resolved? */
+	isResolving: boolean;
+
+	/** Is the record resolved by now? */
+	hasResolved: boolean;
+
+	/** Were there eny edits applied to this entity record? */
+	hasEdits: boolean;
+
+	/** Resolution status */
+	status: Status;
+}
+
+/**
+ * Resolves the specified entity record.
+ *
+ * @param {string} kind     Kind of the deleted entity.
+ * @param {string} name     Name of the deleted entity.
+ * @param {string} recordId Record ID of the deleted entity.
+ *
+ * @example
+ * ```js
+ * import { useEntityRecord } from '@wordpress/core-data';
+ *
+ * function PageTitleDisplay( { id } ) {
+ *   const { record, isResolving } = useEntityRecord( 'postType', 'page', id );
+ *
+ *   if ( isResolving ) {
+ *     return 'Loading...';
```

Comment:
This made me think that maybe at this level of abstraction it could better match the UI feedback so the name of the property could be `isLoading` and the status for the enum `Loading`. I don’t feel strongly about it, so you can ignore it if you don’t like the idea.

Reply:
I also like the name `isLoading` more but I guess @adamziel deliberately wants to match the current convention of _resolving_ data 😅.

Reply:
It is my pet peeve – I'd love to call this `isLoading`! My concern is that we say `resolving` everywhere in `@wordpress/data` and `@wordpress/core-data` and I'd rather stick to that convention instead of introducing a new term for the same concept. My thinking is that I'd rather always say `resolving` than have moments like _ughh is it loading or resolving here? I never rememeber_.

I even thought about deprecating the word `resolving` for loading, but it makes sense that `resolvers` resolve, not load the data.